zebra-chain/src/parallel/tree.rs

```diff
@@ -44,37 +52,35 @@ impl NoteCommitmentTrees {
     ///
     /// If any of the tree updates cause an error,
     /// it will be returned at the end of the parallel batches.
-    #[allow(clippy::unwrap_in_result)]
     pub fn update_trees_parallel(
         &mut self,
         block: &Arc<Block>,
     ) -> Result<(), NoteCommitmentTreeError> {
-        self.update_trees_parallel_list(
-            [(
-                block
-                    .coinbase_height()
-                    .expect("height was already validated"),
-                block.clone(),
-            )]
-            .into_iter()
-            .collect(),
-        )
+        self.update_trees_parallel_list(block.clone())
     }
 
     /// Updates the note commitment trees using the transactions in `block`,
     /// then re-calculates the cached tree roots, using parallel `rayon` threads.
     ///
     /// If any of the tree updates cause an error,
     /// it will be returned at the end of the parallel batches.
+    #[allow(clippy::unwrap_in_result)]
     pub fn update_trees_parallel_list(
         &mut self,
-        block_list: BTreeMap<Height, Arc<Block>>,
+        block: Arc<Block>,
     ) -> Result<(), NoteCommitmentTreeError> {
+        let height = block
+            .coinbase_height()
+            .expect("height was already validated");
+        let block_list: BTreeMap<Height, Arc<Block>> = [(height, block)].into_iter().collect();
+
         // Prepare arguments for parallel threads
         let NoteCommitmentTrees {
             sprout,
             sapling,
             orchard,
+            sapling_subtree: _,
+            orchard_subtree: _,
```

Comment:
```suggestion
            ..
```

Reply:
Applied in https://github.com/ZcashFoundation/zebra/pull/7334/commits/5dd93ec4f0cd3eb3da9bac87b5bf751e72cfc780.